src/plugins/intel_gpu/src/graph/broadcast.cpp

```diff
@@ -95,6 +95,77 @@ std::vector<layout> broadcast_inst::calc_output_layouts(broadcast_node const& /*
 
 template std::vector<layout> broadcast_inst::calc_output_layouts<ov::PartialShape>(broadcast_node const& node, const kernel_impl_params& impl_param);
 
+kernel_impl_params broadcast_inst::get_agnostic_updated_params(kernel_impl_params const& orig_impl_param) {
```

Comment:
BTW can this special handling be done in extend_to_6d  like gemm and shape_of? 

If we add a new additional method like this, I think we should have a common handling of shape for gemm, shape_of, fc, etc.. 
Currently 1)  gemm and shape_of are handled hard coded way in extend_to_6d
while 2) this broadcast is handled by this function. 

Then can we consider to have extend_to_6d function as this? 
e.g., have a common extend_to_6d function by defautl, + overrided extend_to_6d funciton for special cases like this.

Reply:
Refactored to extend_to_6d